port/main.c

```diff
@@ -105,6 +105,7 @@ static int writeStatus(char *buffer, size_t bufSize, size_t offset)
 }
 
 
+#if HAVE_PF
```

Comment:
I would suggest `#ifdef` or `#if defined(HAVE_PF)` as we're probably not compiling with `-Wundef`.

I would be happy to say that with `HAVE_*` macros, `#ifdef` is usually used, buuut - I've checked my `/usr/include` and - usually it's true (`#ifdef` for one `HAVE_*` flag, `#if defined(XXX) && defined(YYY)` for more complex ones), but there are some notable exceptions in well-known files (like `math.h`, or `stdlib.h`) - so I'm not forcing my point of view :).

Also: the support of HAVE_PF should be straightened up:
 - I see it's defined in `lwipopts.h` (I assume it should be done there, not in CFLAGS?), but this file is not explicitly included in the `port/main.c` (but included in other places checking for HAVE_PF)
 - other files use `#ifdef HAVE_PF` so either go back to `#ifdef` or switch to `#if everywhere`

Reply:
As I told before if option is setted in lwipopts.h then it should be  #if something 
Agree that HAVE_PF will not be defaulted in lib_lwip BUT i think shouldn't be different handled than other option.

I enabled -Wundef after previous change :) 

IDK if this should be defined here instead of CFLAGS

Reply:
Fine by me, sorry I forgot the `-Wundef` was already added.

I think we should maintain some convention, eg. options which are to be defined in `lwipopts.h` should be prefixed with `LWIP_` (or `LWIP_EXT_` if this our extension - like `LWIP_EXT_PF`?), all other `HAVE_*` defines are passed from Makefile (via `CFLAGS`).

Maybe it would be good to change it right now but, it's not a blocker against merging (as this topic is partially unrelated).

Please resolve this conversation if You don't want to change it.

Reply:
TBH I think we have a big inconsistency – drivers defines by env and features changes by file lwipopts. I like LWIP_EXT_* so I changed it but I think we should define desired solution and make steps to do things in that way – maybe by that small changes also :)

Reply:
Yeah, would be great to maintain some consistency. I hope that we will come to some conclusions when we will be using lwIP in many projects simultaneously (with different drivers and feature sets) - and come up with better solution then.